apps/src/javalab/Javalab.js

```diff
@@ -76,6 +80,15 @@ Javalab.prototype.init = function(config) {
   const onContinue = this.onContinue.bind(this);
   const onCommitCode = this.onCommitCode.bind(this);
   const onInputMessage = this.onInputMessage.bind(this);
+  let visualization;
+  if (this.level.csaViewMode === CsaViewMode.NEIGHBORHOOD) {
+    const miniApp = new Neighborhood();
+    config.afterInject = () =>
+      miniApp.afterInject(this.level, this.skin, config, this.studioApp_);
+    visualization = <MazeVisualization />;
+  } else {
+    visualization = <DefaultVisualization />;
+  }
```

Comment:
I'd be very interested to see logic like this pulled out into like a `JavalabVisualizationWrapper` class. It could take `this.level` as argument, and could be the one place in the codebase that contains all the knowledge necessary about which properties of level matter for visualization and how to parse them.

Reply:
It could also be responsible for defining the interface that other components use to interact with the visualization, ideally in an abstract enough way that the different underlying visualizations (Maze, Default, etc) could then each define their own implementation for those things that are distinct between them

Reply:
It could maybe even define initialization functionality that gets invoked by `loadJavalab`